src/frequenz/channels/util/_file_watcher.py

```diff
@@ -24,10 +27,19 @@ class EventType(Enum):
         MODIFY = Change.modified
         DELETE = Change.deleted
 
+    @dataclass(frozen=True)
+    class Event:
+        """A file change event."""
+
+        type: FileWatcher.EventType
+        """The type of change that was observed."""
+        path: pathlib.Path
+        """The path where the change was observed."""
+
     def __init__(
         self,
-        paths: List[Union[pathlib.Path, str]],
-        event_types: Optional[Set[EventType]] = None,
+        paths: list[pathlib.Path | str],
+        event_types: set[EventType] | None = None,
```

Comment:
Might it be better to remove the `None` part and just default-init it with `set(FileWatcher.EventType)`? Then we can remove the `if` in line `51`

Reply:
That's not a bad idea, but we would need to use `frozenset(FileWatcher.EventType)` then, because `set` is mutable and [the problem with mutable default arguments in Python](https://editorconfig.org/). If we use `frozenset()` for the default, then `event_types` type should be `collections.abc.Set` (or maybe `collections.abc.Iterable` would be enough if we convert it to a `frozenset` internally, then the argument type can be `Iterable` and the attribute type `frozenset`, so it can't be modified either via the attribute).

Reply:
created the issue https://github.com/frequenz-floss/frequenz-channels-python/issues/88